### PR TITLE
Hocs 2583 request pit extension

### PR DESCRIPTION
**Pull request by agraham19**
* Add endpoints to apply extensions to case
* Add db tables to hold extensions applied to case

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataResource.java [hunk 1]
```diff
@@ -82,6 +87,21 @@ ResponseEntity updateCaseData(@PathVariable UUID caseUUID, @PathVariable UUID st
         return ResponseEntity.ok().build();
     }
 
+    @Authorised(accessLevel = AccessLevel.WRITE)
+    @PutMapping(value = "/case/{caseUUID}/stage/{stageUUID}/extension")
```
Comment (SeanReedHomeOffice) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataResource.java:91]: I think this should be a POST as we are create a brand new extension.

File: src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDeadlineExtensionTypeRepository.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.digital.ho.hocs.casework.domain.repository;
```
Comment (SeanReedHomeOffice) [inline src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDeadlineExtensionTypeRepository.java:1]: Some unused imports here.

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/ApplyExtensionRequest.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.digital.ho.hocs.casework.api.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+public class ApplyExtensionRequest {
```
Comment (SeanReedHomeOffice) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/ApplyExtensionRequest.java:9]: I have a slight concern that for some future tenant we might want an extension with a customisable period of days. Not sure if that is likely.

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java [hunk 4]
```diff
@@ -1,7 +1,12 @@
 package uk.gov.digital.ho.hocs.casework.api;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
```
Comment (SeanReedHomeOffice) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java:4]: Unused imports

**Review by SeanReedHomeOffice — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/ApplyExtensionRequest.java (hunk 3 above)
Comment (agraham19) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/ApplyExtensionRequest.java:9]: The type refers to a type in casework.case_deadline_extension_type schema. This keeps a record of the type name and the number of days assigned to it. We can create new extension types by adding a record into this db with the name and the number of days.

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataResource.java (hunk 1 above)
Comment (agraham19) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataResource.java]: I have made this change

File: src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDeadlineExtensionTypeRepository.java (hunk 2 above)
Comment (agraham19) [reply src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDeadlineExtensionTypeRepository.java:1]: I have addressed this

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java (hunk 4 above)
Comment (agraham19) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java]: I have addressed this

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/ApplyExtensionRequest.java (hunk 3 above)
Comment (SeanReedHomeOffice) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/ApplyExtensionRequest.java:9]: I do get that, I was just wondering if more generally there could be a type of extension where the number of days extended is not set in stone, but could be passed as a parameter. Probably that is unlikely so I reckon your approach is good.

**Review by SeanReedHomeOffice — APPROVED** [review]
LGTM

**Review by SeanReedHomeOffice — APPROVED** [review]
LGTM

**Pull request merged**